libs/cpp-ledger/ledger/utils.cpp

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "cpp-ledger/ledger/utils.hpp"
+
+#include <wchar.h>
+#include <codecvt>
+#include <cstring>
+#include <locale>
+
+namespace ledger {
+
+  void copyStr(char *dest, const char *src) {
+    if (src == nullptr) {
+      return;
+    }
+
+    if (dest != nullptr) {
+      delete[] dest;
+    }
+
+    dest = new char[std::strlen(src) + 1];
```

Comment:
memory leak!

Reply:
As I understand there is no memleak, because it is used only in `DeviceHid` and values are deleted in destructor, But it is bug-prone code. I suggest to create RAI version of `hid_device_info` and delete this code.

Reply:
Yes, there is no memory leak.
The idea is that *dest is a pointer to string. This code is used to copy one string to another. At the previous step we cleared dest and then allocated memory for new string. This memory should stay allocated after this method.
@Alexey-N-Chernyshov is right, finally this memory is deleted in destructor of DeviceHid.

I will think about other solution